### ISSUE TITLE
Can't import some class in TypeScript

### ISSUE THREAD

**Issue by ppodds** [psibean/discord.js-pagination#13]
Lack of typing declares cause this problem.
It can be fixed by adding missing typing declares in `index.d.ts`.

tsc version: 4.5.4
package version: 4.0.5

**Comment by psibean** [comment]
Yep. This is a duplicate of issue #9, typings support is needed.

I've given it a shot, but I seem to encounter some unnecessary design complexity around the pageIdentifiers object typing and the collectorOptions typing as the filter changes depending on the paginator. And also on the users specific identifiers object.

I've been short on time to re-visit this, but also happy for others to give it a shot and PR.
